h5py/_hl/compat.py

```diff
@@ -0,0 +1,79 @@
+"""
+Compatibility module for high-level h5py
+"""
+import sys
+import six
+
+
+def fspath(path):
```

Comment:
see my other comment in `files.py`. perhaps we can do: if python version < 3.6 `def fspath ...` else `from os import fspath`? analogously for fsdecode an fsencode (for python < 3.2)? then we just do a single `from .compat import fspath`. hope it make sense